editoast/src/views/temporary_speed_limits.rs

```diff
@@ -0,0 +1,268 @@
+use crate::Create;
+use crate::CreateBatch;
+use crate::Model;
+use editoast_authz::BuiltinRole;
+
+use crate::AppState;
+
+crate::routes! {
+    "/temporary_speed_limits" => create,
+}
+
+editoast_common::schemas! {
+    TemporarySpeedLimitCreateForm,
+    TemporarySpeedLimitCreateResponse,
+    TemporarySpeedLimitItemForm,
+}
+
+#[derive(Serialize, ToSchema)]
+struct TemporarySpeedLimitItemForm {
+    start_date_time: NaiveDateTime,
+    end_date_time: NaiveDateTime,
+    track_ranges: Vec<DirectionalTrackRange>,
+    speed_limit: f64,
+    obj_id: String,
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+struct TemporarySpeedLimitCreateForm {
+    speed_limit_group_name: String,
+    speed_limits: Vec<TemporarySpeedLimitItemForm>,
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+struct TemporarySpeedLimitCreateResponse {
+    pub group_id: i64,
+}
+
+impl TemporarySpeedLimitItemForm {
+    pub fn into_temporary_speed_limit_changeset(
+        self,
+        temporary_speed_limit_group_id: i64,
+    ) -> Changeset<TemporarySpeedLimit> {
+        TemporarySpeedLimit::changeset()
+            .start_date_time(self.start_date_time)
+            .end_date_time(self.end_date_time)
+            .track_ranges(self.track_ranges)
+            .speed_limit(self.speed_limit)
+            .obj_id(self.obj_id)
+            .temporary_speed_limit_group_id(temporary_speed_limit_group_id)
+    }
+}
+
+impl<'de> Deserialize<'de> for TemporarySpeedLimitItemForm {
+    fn deserialize<D>(deserializer: D) -> StdResult<TemporarySpeedLimitItemForm, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct Internal {
+            start_date_time: NaiveDateTime,
+            end_date_time: NaiveDateTime,
+            track_ranges: Vec<DirectionalTrackRange>,
+            speed_limit: f64,
+            obj_id: String,
+        }
+        let Internal {
+            start_date_time,
+            end_date_time,
+            track_ranges,
+            speed_limit,
+            obj_id,
+        } = Internal::deserialize(deserializer)?;
+
+        // Check dates
+        if end_date_time <= start_date_time {
+            return Err(SerdeError::custom(format!(
+                "The temporary_speed_limit start date '{}' must be before the end date '{}'",
+                start_date_time, end_date_time
+            )));
+        }
+
+        Ok(TemporarySpeedLimitItemForm {
+            start_date_time,
+            end_date_time,
+            track_ranges,
+            speed_limit,
+            obj_id,
+        })
+    }
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "temporary_speed_limit")]
+pub enum WorkScheduleError {
+    #[error("Name '{name}' already used")]
+    #[editoast_error(status = 400)]
+    NameAlreadyUsed { name: String },
+}
+
+pub fn map_diesel_error(e: InternalError, name: impl AsRef<str>) -> InternalError {
```

Comment:
```suggestion
fn map_diesel_error(e: InternalError, name: impl AsRef<str>) -> InternalError {
```

Once we decide how to handle errors in editoast, we'll have to rethink how we forward pg errors in ModelV2's API. Ideally the constraint name matching should be done in `editoast_models`. Nothing you can do about it rn though, feel free to resolve.